src/core/patch.rkt

```diff
@@ -71,16 +71,27 @@
                               #;(exp ,exp-x ,log-x)
                               #;(log ,log-x ,exp-x))))
 
-(define (taylor-alt altn)
-  (define expr (prog->spec (alt-expr altn)))
+(define (taylor-alts altns)
+  (define exprs
+    (for/list ([altn (in-list altns)])
+      (prog->spec (alt-expr altn))))
+  (define free-vars (map free-variables exprs))
```

Comment:
This is fine and we don't have to do it now, but interestingly both `prog→spec` and `free-variables` could be batchified as well.